cmake/modules/hunter_experimental_add_host_subdirectory.cmake

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2016 Fabian Winnen
+# All rights reserved.
+
+# Adds a subdirectory with CMakeList.txt as seperate build with default (host) toolchain
+
+# Usage:
+#   hunter_add_host_subdirectory(host)
+#   hunter_add_host_subdirectory(host
+#       BINARY_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/host-build
+#       ENV_VARS CC CXX
```

Comment:
> I don't think that we should provide interface for user so he can set list of environment variables. 

This was meant as a experimental feature to fix problems for users which I did not encounter, since I do not fully understand whats causing the problem

> Also I'm not sure about using clear-all.cmake module since usually this module used in a separate cmake process so it doesn't affect user's environment. You can try it and if there will be no problems do include it.

Do you mean including clear-all.cmake on client side in host/CMakeList.txt before project (I think that is too late for cmake not to pick that false CC up) or in hunter_experimental_add_host_subdirectory before `execute_process`. The Problem I see with the later is, that it may clear some variables purposefully set by some toolchain and not restore them afterwards.

> Same with binary directory, see the notes below.

I will set a fixed binary_directory for the host project as you recommended. But I don't understand what you mean with: 

> By the way it will be convenient if we do return HUNTER_HOST_ROOT variable to the parent scope so user don't have to read our _3rdParty/Hunter/... files.